### ISSUE TITLE
ActionType mismatch

### ISSUE THREAD

**Issue by khueue** [joshforisha/cycle-firebase#16]
Hi!

I've been fiddling a bit with cycle.js and cycle-fire recently and I couldn't seem to get the first example (sourcing and sinking a number) in your readme to work! The sourcing went fine, and manual changes to the database through the Firebase console got reflected immediately. Sinking, on the other hand, didn't propagate to Firebase. The number did not change, and no errors showed up.

Pardon my debugging, since my TypeScript skills are next to none. What I did was to augment both of the `addListener` calls in driver.js (yes, the generated JS file) like this (and I'm guessing there's a much better way of doing this...):

```
        ....addListener({
            complete: function () {},
            error: function (x) { console.error('ERROR', x) },
            next: function () {},
        });
```

When then clicking on the shuffle button, to trigger the `set` action, I got a message like the following:

```
{
    code: "auth/argument-error",
    message: "confirmPasswordReset failed: First argument "code" must be a valid string."
}
```

Looking at the enums and the ActionType declaration, I found that they share number designations:

```
    AuthActionType[AuthActionType["ApplyActionCode"] = 0] = "ApplyActionCode";
    AuthActionType[AuthActionType["CheckActionCode"] = 1] = "CheckActionCode";
    AuthActionType[AuthActionType["ConfirmPasswordReset"] = 2] = "ConfirmPasswordReset";
    ...
    ReferenceActionType[ReferenceActionType["Push"] = 0] = "Push";
    ReferenceActionType[ReferenceActionType["Remove"] = 1] = "Remove";
    ReferenceActionType[ReferenceActionType["Set"] = 2] = "Set";
```

Just changing these manually so that **each number is unique** fixes the issue and everything works as expected! Like I said, my TypeScript foo sucks, so I'm unable to provide you with a PR, but hopefully this will guide you in the right direction.

Looking forward to hearing from you!

**Comment by joshforisha** [comment]
Hey, thanks a ton for the investigation. I've gone back and forth with how to type the actions properly, and made some assumptions about how TS transpiles union enums which have turned out to be incorrect. You stumbled upon the problem in a way I probably should have.

Anyway, I'll be trying some conversions and see which method actually works as I've intended and give updates here. Thanks again.

**Comment by joshforisha** [comment]
Alright, so I just renamed the base TS `ActionType`s and collapsed them into a single enum to fix the problem in `v2.3.0`. Arguably a tad messier, but it fixes the problem.

Thanks again for the research and issue.

**Comment by khueue** [comment]
Sweet, works now!
